front/pages/w/[wId]/u/chat/[cId]/index.tsx

```diff
@@ -679,207 +679,212 @@ export default function AppChat({
     }
   };
 
-  const handleSubmit = async () => {
-    let runRetrieval = true;
-    let runAssistant = true;
-    let processedInput = input;
+  const runChatRetrieval = async (m: ChatMessageType[], query: string) => {
+    const retrievalMessage: ChatMessageType = {
+      role: "retrieval",
+    };
+    setResponse(retrievalMessage);
 
-    if (input.startsWith("/retrieve")) {
-      runAssistant = false;
-      processedInput = input.substring("/retrieve".length).trim();
+    const config = cloneBaseConfig(
+      DustProdActionRegistry["chat-retrieval"].config
+    );
+    config.DATASOURCE.data_sources = dataSources
+      .filter((ds) => ds.selected)
+      .map((ds) => {
+        return {
+          workspace_id: prodAPI.workspaceId(),
+          data_source_id: ds.name,
+        };
+      });
+    if (selectedTimeRange.id !== "all") {
+      config.DATASOURCE.filter = {
+        timestamp: { gt: Date.now() - selectedTimeRange.ms },
+      };
     }
-    if (input.startsWith("/follow-up")) {
-      runRetrieval = false;
-      processedInput = input.substring("/follow-up".length).trim();
+    const res = await runActionStreamed(owner, "chat-retrieval", config, [
+      {
+        messages: [{ role: "query", message: query }],
+        userContext: {
+          timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone,
+          localeString: navigator.language,
+        },
+      },
+    ]);
+    if (res.isErr()) {
+      m.push({
+        role: "error",
+        message: res.error.message,
+      } as ChatMessageType);
+      // console.log("ERROR", res.error);
+      setMessages(m);
+      setResponse(null);
+      setLoading(false);
+      return;
```

Comment:
beware that return would skip the runAssistant if retrieval had failed. You might runRetrieval now right?